lib/src/charts/interactors/cartesian/bar_hit_tester.dart

```diff
@@ -0,0 +1,444 @@
+        interactionGroup = _findNearestGroupByDistance(
+          localPosition: localPosition,
+          previousGroup: previousGroup,
+          currentGroup: currentGroup,
+        );
+        break;
+      case SnapToBarBehaviour.snapToSection:
+        interactionGroup = _findNearestGroupBySection(
+          graphUnitWidth: graphUnitWidth,
+          localPosition: localPosition,
+          previousGroup: previousGroup,
+          currentGroup: currentGroup,
+        );
+        break;
+    }
+
+    return _getBarInteraction(
+      group: interactionGroup,
+      groupType: interactionGroup.type,
+      localPosition: localPosition,
+      snapToBarConfig: snapToBarConfig,
+      fuzziness: fuzziness,
+      type: type,
+    );
+  }
+
+  static BarInteractionResult? _getBarInteraction({
+    required BarGroupInteractionData group,
+    required InteractedGroupType groupType,
+    required Offset localPosition,
+    required SnapToBarConfig snapToBarConfig,
+    required Fuzziness fuzziness,
+    required TouchInteractionType type,
+  }) {
+    switch (groupType) {
+      case InteractedGroupType.simpleBar:
+        return _getSeriesBarInteraction(
+          group: group,
+          localPosition: localPosition,
+          snapToBarConfig: snapToBarConfig,
+          fuzziness: fuzziness,
+          type: type,
+        );
+      case InteractedGroupType.multiBarSeries:
+        return _getSeriesBarInteraction(
+          group: group,
+          localPosition: localPosition,
+          snapToBarConfig: snapToBarConfig,
+          fuzziness: fuzziness,
+          type: type,
+        );
+      case InteractedGroupType.multiBarStack:
+        return _getStackedBarInteractionResult(
+          group: group,
+          localPosition: localPosition,
+          snapToBarConfig: snapToBarConfig,
+          fuzziness: fuzziness,
+          type: type,
+        );
+    }
+  }
+
+  static BarInteractionResult? _getSeriesBarInteraction({
+    required BarGroupInteractionData group,
+    required Offset localPosition,
+    required SnapToBarConfig snapToBarConfig,
+    required Fuzziness fuzziness,
+    required TouchInteractionType type,
+  }) {
+    final bars = group.barInteractions;
+    BarInteractionData? previousBar;
+    for (var i = 0; i < bars.length; i++) {
+      final bar = bars[i];
+      final shouldSnapToHeight = snapToBarConfig.shouldSnapToHeight(type);
+
+      if (bar.containsWithFuzziness(
+        position: localPosition,
+        fuzziness: fuzziness,
+        snapToHeight: shouldSnapToHeight,
+      )) {
+        return bar.getInteractionResult(localPosition, type);
+      }
+
+      if (snapToBarConfig.shouldSnapToWidth(type)) {
+        final isLastBar = i == bars.length - 1;
+        final isPointerAfterBar = bar.isPointerAfterBar(localPosition);
+        // The condition here is
+        // 1. If Pointer is after bar and
+        // 2. Is not the last bar.
+        if (!isPointerAfterBar || isLastBar) {
```

Comment:
This comment is incorrect. 
Condition is if the pointer is not after the bar or this is the last bar.

And the condition seems simple enough, I don't think we need a comment to explain it.